cmake/modules/FindLibsrtp.cmake

```diff
@@ -0,0 +1,79 @@
+#.rst:
+# FindLibsrtp
+# -------
+# Finds the libsrtp library
+#
+# This will define the following variables::
+#
+#   LIBSRTP_FOUND - system has libsrtp
+#   LIBSRTP_INCLUDE_DIRS - the libsrtp include directory
+#   LIBSRTP_LIBRARIES - the libsrtp libraries
+#
+# and the following imported targets::
+#
+#   libsrtp::libsrtp - The libsrtp library
+#
+
+# If target exists, no need to rerun find. Allows a module that may be a
+# dependency for multiple libraries to just be executed once to populate all
+# required variables/targets.
+if(NOT TARGET libsrtp::libsrtp)
+  if(ENABLE_INTERNAL_LIBSRTP)
+    include(cmake/scripts/common/ModuleHelpers.cmake)
+
+    set(MODULE_LC libsrtp)
+
+    SETUP_BUILD_VARS()
+
+    set(LIBSRTP_VERSION ${${MODULE}_VER})
+
+    set(CMAKE_ARGS -DBUILD_WITH_WARNINGS=OFF
+                   -DENABLE_OPENSSL=ON
+                   -DLIBSRTP_TEST_APPS=OFF)
+
+    set(patches "${CORE_SOURCE_DIR}/tools/depends/target/${MODULE_LC}/0001-Disable-build-with-warnings.patch"
+                "${CORE_SOURCE_DIR}/tools/depends/target/${MODULE_LC}/0002-Enable-OpenSSL.patch")
+
+    generate_patchcommand("${patches}")
+
+    BUILD_DEP_TARGET()
+  else()
+    # Populate paths for find_package_handle_standard_args
+    find_path(LIBSRTP_INCLUDE_DIR NAMES srtp2/srtp.h)
+    find_path(LIBSRTP_LIBRARY NAMES srtp2)
+  endif()
+
+  include(FindPackageHandleStandardArgs)
+  find_package_handle_standard_args(Libsrtp
+    REQUIRED_VARS
+      LIBSRTP_LIBRARY
+      LIBSRTP_INCLUDE_DIR
+    VERSION_VAR
+      LIBSRTP_VERSION)
+
+  if(LIBSRTP_FOUND)
+      set(LIBSRTP_INCLUDE_DIRS "${LIBSRTP_INCLUDE_DIR}")
+      set(LIBSRTP_LIBRARIES "${LIBSRTP_LIBRARY}")
```

Comment:
If you are going to use the target, dont set these. Will just make my life easier as it gets phased out.

Reply:
This will be fixed when me migrate this PR to target usage similar to https://github.com/xbmc/xbmc/pull/23612.